tests/Doctrine/Tests/TestUtil.php

```diff
@@ -43,7 +53,31 @@ class TestUtil
      */
     public static function getConnection(): Connection
     {
-        $conn = DriverManager::getConnection(self::getConnectionParams());
+        $params = self::getConnectionParams();
+        $conn   = DriverManager::getConnection($params);
+        // Note, writes direct to STDERR to prevent phpunit detecting output - otherwise this would cause either an
+        // "unexpected output" warning or a failure on the first test case to call this method.
+        fwrite(
+            STDERR,
+            sprintf(
+                "\nUsing DB driver %s (from %s connection params)\n",
+                get_class($conn->getDriver()),
+                $params['is_fallback'] ? 'fallback' : 'specified'
+            )
+        );
+
+        $expectDriver = getenv('EXPECT_DB_DRIVER');
```

Comment:
Maybe we should throw if the `CI` env var is true and this is not set? That way if we migrate to another CI system, we will have to implement that `EXPECT_DB_DRIVER` variable

Reply:
@greg0ire only thing is the `CI` env var is also just set by github actions so not necessarily guaranteed it would be present on a new CI system...?

The other possibility is to just remove the concept of the `fallback` connection, and require that the phpunit.xml you're using specifies a driver and connection details. The default phpunit.xml.dist (and the `sqlite.xml` used in CI) would just need to have explicit config for the sqlite db.

Since it'd be much harder for that to go wrong, the extra `EXPECT_DB_DRIVER` check might then not be required, which would reduce some duplication / coupling between the workflow files and the phpunit configs? Probably still worth displaying which driver was selected for debugging / visual confirmation if changing CI system.

That does diverge from how dbal do it - don't know how maintainers are working across projects / how important it is to have consistency?

Reply:
>  only thing is the CI env var is also just set by github actions so not necessarily guaranteed it would be present on a new CI system...?

Isn't it a standard? Because it looks a lot like it is:

https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci1/environment-variables.md
https://www.appveyor.com/docs/environment-variables/

> Since it'd be much harder for that to go wrong

It would be, but I like the idea of checking that kind of expectation a lot. In fact I thought a bit about it some time ago, because I was not fully at ease with the concept of skipped tests: how can we be sure that skipped tests are run on at least one job? Maybe there should be a coverage report for tests themselves and it should always be 100% ? I don't know if that's even possible. And I think that `CI` env var would help us being lenient on dev, and super strict on the CI.

> the extra EXPECT_DB_DRIVER check might then not be required, which would reduce some duplication / coupling between the workflow files and the phpunit configs?

I wonder if it might be possible to reference a matrix variable in the env section, so as not to have that much duplication. I'll try to push something to your branch soon.

> That does diverge from how dbal do it - don't know how maintainers are working across projects / how important it is to have consistency?

I'm not a developer as much as a maintainer in Doctrine, so I can answer that question: to me it is quite important, because many of the contributions I do to Doctrine are org-wide (and can be followed here: https://github.com/orgs/doctrine/projects/7). Having one way of doing things is better for me, and I think it is also better for contributors. Let's draft something and have both @beberlei and @morozov sign off on it.

Reply:
>  I'll try to push something to your branch soon.

Or rather, I'll try on my fork since I seem to lack the necessary permissions

Reply:
> Isn't it a standard? Because it looks a lot like it is:

Ah, I hadn't realised. Sounds good.

> possible to reference a matrix variable in the env section

Yes, I think not possible in the overall `env` (not least as not all the jobs have a matrix.extension value) but should work at job level which is less duplication than I had.